command/fmt_test.go

```diff
@@ -0,0 +1,106 @@
+package command
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/nomad/ci"
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFmtNomadJob(t *testing.T) {
+	ci.Parallel(t)
+
+	ui := cli.NewMockUi()
+
+	cmd := &FormatCommand{
+		Meta: Meta{Ui: ui},
+	}
+
+	if code := cmd.Run([]string{"testdata/example-basic.nomad", "testdata/example-vault.nomad"}); code != 0 {
+		t.Fatalf("expected code 0, got %d", code)
+	}
+}
+
+func TestFmtNomadJobDontOverwrite(t *testing.T) {
+	ci.Parallel(t)
+
+	ui := cli.NewMockUi()
+	cmd := &FormatCommand{
+		Meta: Meta{Ui: ui},
+	}
+
+	assert.Equal(t, 0, cmd.Run([]string{"-overwrite=false", "testdata/example-basic.nomad"}))
+
+	fileBytes, err := os.ReadFile("testdata/example-basic.nomad")
+	require.NoError(t, err)
+
+	assert.Contains(t, ui.OutputWriter.String(), string(fileBytes))
+}
+
+func TestFmtFileDoesNotExist(t *testing.T) {
+	ci.Parallel(t)
+
+	ui := cli.NewMockUi()
+	cmd := &FormatCommand{
+		Meta: Meta{Ui: ui},
+	}
+
+	assert.Equal(t, 1, cmd.Run([]string{"file/does/not/exist.hcl"}))
+}
+
+func TestFmtWrongHCLFile(t *testing.T) {
+	ci.Parallel(t)
+
+	ui := cli.NewMockUi()
+
+	cmd := &FormatCommand{
+		Meta: Meta{Ui: ui},
+	}
+
+	bytes, err := os.ReadFile("testdata/fmt/bad-nomad.hcl")
+	require.NoError(t, err)
+
+	f, err := os.CreateTemp("", "nomad-fmt-")
```

Comment:
This leaves temp files on the test host when we're done. You can use [`t.TempDir`](https://pkg.go.dev/testing#T.TempDir) to get a directory that'll automatically get cleaned up afterwards.